app/ide-desktop/lib/client/src/index.ts

```diff
@@ -47,16 +72,57 @@ class App {
     isQuitting = false
 
     async run() {
-        const { args, windowSize, chromeOptions } = configParser.parseArgs()
-        this.args = args
+        // On macOS when Enso-associated file is opened, the application is first started and then it
+        // receives the `open-file` event. However, if there is already an instance of Enso running,
+        // it receives the `open-file` event (and no new instance is created for us). In this case,
+        // we manually start a new instance of the application and pass the file path to it (using the
+        // Windows-style command).
+        electron.app.on('open-file', (event, path) => {
```

Comment:
I think this whole function should be in a separate module, like `client/src/file-assotiations` and it should just export function `handleFileAssociations` or something like that. This function should be called from `index`. We try to keep index as small as possible. It's too big right now, but we are getting there.